clippy_lints/src/collapsible_if.rs

```diff
@@ -81,6 +98,14 @@ impl EarlyLintPass for CollapsibleIf {
     }
 }
 
+impl EarlyLintPass for CollapsibleIfElse {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &ast::Expr) {
+        if !expr.span.from_expansion() {
+            check_if(cx, expr)
+        }
+    }
```

Comment:
Why implement a new lint pass for this. Allowing lints is only dependent on the lint name, not the lint pass.